src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -214,21 +217,27 @@ public override void Start()
         /// </summary>
         private void StartAddressManager(NodeConnectionParameters connectionParameters)
         {
-            this.addressManager = new PeerAddressManager(this.asyncLoopFactory, this.dataFolder.AddressManagerFilePath);
-            connectionParameters.TemplateBehaviors.Add(new PeerAddressManagerBehaviour(this.addressManager));
+            this.addressManager = new PeerAddressManager(this.dataFolder);
+            var addressManagerBehaviour = new PeerAddressManagerBehaviour(this.addressManager);
+            connectionParameters.TemplateBehaviors.Add(addressManagerBehaviour);
 
-            if (File.Exists(Path.Combine(this.dataFolder.AddressManagerFilePath, PeerAddressManager.PEERFILENAME)))
+            if (File.Exists(Path.Combine(this.dataFolder.AddressManagerFilePath, PeerAddressManager.PeerFileName)))
             {
                 this.logger.LogInformation($"Loading peers from : {this.dataFolder.AddressManagerFilePath}...");
                 connectionParameters.PeerAddressManager().LoadPeers();
             }
 
-            this.logger.LogInformation("Starting peer discovery...");
-            connectionParameters.PeerAddressManagerBehaviour().DiscoverPeers(this.network, this.connectionManager.Parameters);
+            if (addressManagerBehaviour.Mode.HasFlag(PeerAddressManagerBehaviourMode.Discover))
```

Comment:
Looks like you can inject `PeerDiscoveryLoop` and just call start at this point.
The if condition can be inside the loop right?

Reply:
Also, did you mean inject it in the constructor of BaseFeature?

I can move the If condition but what if  PeerAddressManagerBehaviourMode is None? Then we have an instance of PeerDiscoveryLoop that we dont use?

Reply:
Hmm yeah, but the loop will stop so its just the instance that will be in memory, is it s a big deal?